### ISSUE TITLE
`npm publish` fails with "operation not permitted" message

### ISSUE THREAD

**Issue by saschanaz** [npm/npm#17818]
#### I'm opening this issue because:

  - [x] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

`npm publish` fails

#### How can the CLI team reproduce the problem?

```
PS C:\Users\표준사용자\Documents\GitHub\glodule> npm publish
npm ERR! path C:\Users\표준사~1\AppData\Local\Temp\npm-6140-d16a26c9\tmp\fromDir-0760baf3\package.tgz
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall unlink
npm ERR! Error: EPERM: operation not permitted, unlink 'C:\Users\표준사~1\AppData\Local\Temp\npm-6140-d16a26c9\tmp\fromDir-0760baf3\package.tgz'
npm ERR!  { Error: EPERM: operation not permitted, unlink 'C:\Users\표준사~1\AppData\Local\Temp\npm-6140-d16a26c9\tmp\fromDir-0760baf3\package.tgz'
npm ERR!   cause:
npm ERR!    { Error: EPERM: operation not permitted, unlink 'C:\Users\표준사~1\AppData\Local\Temp\npm-6140-d16a26c9\tmp\fromDir-0760baf3\package.tgz'
npm ERR!      errno: -4048,
npm ERR!      code: 'EPERM',
npm ERR!      syscall: 'unlink',
npm ERR!      path: 'C:\\Users\\표준사~1\\AppData\\Local\\Temp\\npm-6140-d16a26c9\\tmp\\fromDir-0760baf3\\package.tgz' },
npm ERR!   isOperational: true,
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'unlink',
npm ERR!   path: 'C:\\Users\\표준사~1\\AppData\\Local\\Temp\\npm-6140-d16a26c9\\tmp\\fromDir-0760baf3\\package.tgz' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\표준사용자\AppData\Roaming\npm-cache\_logs\2017-07-18T00_30_51_644Z-debug.log
```

Full log: https://gist.github.com/SaschaNaz/7b6b39008f4217df336adfe8a3918bc1

Running with administrator privilege doesn't help.

### supporting information:

 - `npm -v` prints: 5.0.3
 - `node -v` prints: v8.1.4
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Windows 7
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by g1eny0ung** [comment]
Same issue for me, run with admin but not work.Have you found a solution? @SaschaNaz

**Comment by saschanaz** [comment]
`npm login` before `npm publish` is a workaround. I think npm 4 automatically sent me to login process when I did `npm publish`.

**Comment by g1eny0ung** [comment]
@SaschaNaz Right, `npm publish` is worked after `npm login`.I forgot to run `npm login` after I install npm on  this new windows machine **QAQ**.Thank u for help!

**Comment by gavinr** [comment]
@SaschaNaz thanks for the help and workaround. This was really frustrating!

**Comment by eczn** [comment]
@SaschaNaz thanks for help.  :)

**Comment by wodddy** [comment]
I get the same error on a local installation of npm Enterprise even after trying adduser/login.

`npm whoami` shows my username. `npm config get registry` works correctly. But `npm publish` gives the same error. It worked a week ago though

**Comment by g1eny0ung** [comment]
@wodddy What about re execute `npm login` before `npm publish`?Maybe this can solve your problem.

**Comment by wodddy** [comment]
@g1eny0ung I tried all possible combinations of adduser/login/whoami/publish. I will wait until Monday. Who knows, maybe there will be some update on nom which will help somehow.

**Comment by g1eny0ung** [comment]
@wodddy 
```
[~] cat .npmrc
registry=http://registry.npmjs.org/
electron_mirror=https://npm.taobao.org/mirrors/electron/
//registry.npmjs.org/:_authToken=bd4789c0-xxxxx
```

Do you have the auth token in yours .npmrc?

**Comment by wodddy** [comment]
@g1eny0ung sure. Both for npmjs and private registries

**Comment by g1eny0ung** [comment]
@wodddy All right. I don't know why your npm doesn't work. Maybe it's not npm's problem, try run as admin? Hope you solve this problem. If you solve it, please comment there and everyone can know about it. Actually, I also want to know what causes this problem. 🤒

**Comment by wodddy** [comment]
@g1eny0ung When I run as administrator it still prints the same error. Including a part about
`npm ERR! Please try running this command again as root/Administrator.`

**Comment by leonardoballand** [comment]
@g1eny0ung 's solution worked for me :)

**Comment by 0xffff00** [comment]
@g1eny0ung 's solution works for me .it's very misleading error message.

**Comment by Almenon** [comment]
Also got this error.  Very misleading.

**Comment by Ore4444** [comment]
Suggestion: present the user with a more specific error message in case `npm login` is needed.

**Comment by kespinosa05** [comment]
Use npm login --registry http://<nexusserver:ip>/repository/npm-internal/
